src/pool/srv_pool.c

```diff
@@ -1653,13 +1653,8 @@ read_db_for_stepping_up(struct pool_svc *svc, struct pool_buf **map_buf_out,
 	struct pool_buf	       *map_buf;
 	struct daos_prop_entry *svc_rf_entry;
 	daos_prop_t	       *prop = NULL;
-	uint32_t                svc_ops_enabled = 0;
```

Comment:
The change to this function seems good. No change requested since this form seems cleaner.

Just for confirmation, if the changes to this function were omitted from this patch, the issues in the ticket resulting from the pool upgrade bug would be resolved via the other change in this patch to pool_upgrade_props()?  i.e., For a daos 2.4 pool whose svc_ops_{enabled,age,max} all do not exist, functionally with or without the change to this function, in both cases itt will result in the same svc->ps_ops_enabled = 0, svc->svc_ops_max = 0, svc->svc_ops_age = 0 ?

Reply:
@kccain, yes, I think things should work without the changes in `read_db_for_stepping_up`. Do you prefer not to make these changes?

This question reminds me: To make the code easier to understand, I should have written an `else` branch that sets `enabled`, `max`, and `age` to 0. What do you think?

(Since the CI job has already failed during `git fetch`, I'll refresh the PR to add the `else` branch. But please feel free to share any concern/objection.)

Reply:
No objection, I think your change is an improvement, including the most recent change to add the else branch and improve code readability.